content/articles/python-metaclasses-for-metaprogramming/index.md

```diff
@@ -0,0 +1,233 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /python-metaclasses-for-metaprogramming/
+title: Understanding Metaclasses in Python
+description: This tutorial discusses how to use python's metaclasses for metaprogramming. This process involves writing code that manipulates other code. 
+author: dickson-gitau
+date: 2021-12-23T00:00:00-14:10
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/python-metaclasses-for-metaprogramming/hero.jpg
+    alt: Python Metaclasses for Metaprogramming Hero Image
+---
+Metaprogramming is the practice of writing code that manipulates other code. It refers to the ability of a program to regulate itself.
+<!--more-->
+In Python,  metaprogramming can be implemented using decorators and metaclasses.
+
+Object-oriented programming(OOP) is essential in Python. This concept relies on metaprogramming. 
+
+### Prerequisites
+To follow along, you need some some knowledge of decorators and object-orented programming concepts in Python.
+
+### Table of contents
+- [What are metaclasses](#what-are-metaclasses)
+- [Understanding metaclasses with examples](#understanding-metaclasses-with-examples)
+- [Reasons to use metaclasses instead of functions](#reasons-to-use-metaclasses-instead-of-functions)
+- [Instances to use metaclasses](#instances-to-use-metaclasses)
+- [Customizing metaclasses](#customizing-metaclasses)
+- [Using a metaclass to fix an issue](#using-a-metaclass-to-fix-an-issue)
+- [Conclusion](#conclusion)
+
+### What is a metaclass?
+Metaclasses in Python are classes of classes that specify the behavior of a class. A class is a metaclass in its own right. How the class is implemented is defined by its class definition in Python.

```

Comment:
This point is not clear. I am having trouble understanding. Would you please elaborate?